File: JANUARY/16-01-2026/2975.go
func maximizeSquareArea(m int, n int, hFences []int, vFences []int) int {
	const MOD int64 = 1000000007

	h := make([]int, 0, len(hFences)+2)
	v := make([]int, 0, len(vFences)+2)

	h = append(h, 1, m)
	h = append(h, hFences...)
	v = append(v, 1, n)
	v = append(v, vFences...)

	sort.Ints(h)
	sort.Ints(v)

	hs := make(map[int]struct{})
	for i := 0; i < len(h); i++ {
		for j := i + 1; j < len(h); j++ {
			hs[h[j]-h[i]] = struct{}{}
		}
	}

	best := -1
	for i := 0; i < len(v); i++ {
		for j := i + 1; j < len(v); j++ {
			d := v[j] - v[i]
			if _, ok := hs[d]; ok && d > best {
				best = d
			}
		}
	}

	if best == -1 {
		return -1
	}
	return int((int64(best) * int64(best)) % MOD)
}

File: 2026POTD/JANUARY/16-01-2026/2975.php
class Solution {

    /**
     * @param Integer $m
     * @param Integer $n
     * @param Integer[] $hFences
     * @param Integer[] $vFences
     * @return Integer
     */
    function maximizeSquareArea($m, $n, $hFences, $vFences) {
        $MOD = 1000000007;

        $h = $hFences;
        $v = $vFences;

        $h[] = 1; $h[] = $m;
        $v[] = 1; $v[] = $n;

        sort($h);
        sort($v);

        $hs = [];
        $hlen = count($h);
        for ($i = 0; $i < $hlen; $i++) {
            for ($j = $i + 1; $j < $hlen; $j++) {
                $hs[$h[$j] - $h[$i]] = true;
            }
        }

        $best = -1;
        $vlen = count($v);
        for ($i = 0; $i < $vlen; $i++) {
            for ($j = $i + 1; $j < $vlen; $j++) {
                $d = $v[$j] - $v[$i];
                if (isset($hs[$d])) {
                    if ($d > $best) $best = $d;
                }
            }
        }

        if ($best === -1) return -1;
        return (int)(($best * $best) % $MOD);
    }
}